app/models/vm_or_template/right_sizing.rb

```diff
@@ -163,16 +163,8 @@ def max_mem_usage_absolute_average_max_over_time_period
   end
 
   def cpu_usagemhz_rate_average_max_over_time_period
-    if has_attribute?("cpu_usagemhz_rate_average_max_over_time_period")
-      self["cpu_usagemhz_rate_average_max_over_time_period"]
-    else
-      opts = {
-        :end_date => Time.now.utc,
-        :days     => Metric::LongTermAverages::AVG_DAYS
-      }
-      VimPerformanceAnalysis.find_perf_for_time_period(self, "daily", opts)
-                            .maximum(:cpu_usagemhz_rate_average)
-    end
+    perfs = VimPerformanceAnalysis.find_perf_for_time_period(self, "daily", :end_date => Time.now.utc, :days => Metric::LongTermAverages::AVG_DAYS)
```

Comment:
@jameswnl @gmcculloug 

so before it was going to:
https://github.com/Ladas/manageiq/blob/7350b1de54016b4c356633b6d44068d043c07635/app/models/vm_or_template/right_sizing.rb#L35

and therefore to:
https://github.com/Ladas/manageiq/blob/7350b1de54016b4c356633b6d44068d043c07635/app/models/vm_or_template/right_sizing.rb#L42

and the else branch was never called?

Reply:
----

also we were using SQL max, vs. compact.max in Ruby, I would prefer the sql one https://github.com/Ladas/manageiq/blob/7350b1de54016b4c356633b6d44068d043c07635/app/models/vm_or_template/right_sizing.rb#L45
Probably added by @kbrock? :-)

Reply:
hm, few lines above are similar methods, doing .compact.max, just using different columns

Reply:
@Ladas Did you mean to mention me here or was that for @gtanzillo?  🙏  the latter 😄

Reply:
@gmcculloug yes sorry, it should have been @gtanzillo :-)